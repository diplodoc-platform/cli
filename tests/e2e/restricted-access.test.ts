import {describe, test} from 'vitest';
import {TestAdapter, compareDirectories, getTestPaths} from '../fixtures';

const generateFilesYamlTestTemplate = (
    testTitle: string,
    testRootPath: string
) => {
    test(testTitle, async () => {
        const {inputPath, outputPath} = getTestPaths(testRootPath);
        await TestAdapter.testBuildPass(inputPath, outputPath, {md2md:true});
        await compareDirectories(outputPath);
    });
};

describe('Restricted access', () => {
    generateFilesYamlTestTemplate('Simple restricted access', 'mocks/restricted-access/test1');

    generateFilesYamlTestTemplate('Nested restricted access', 'mocks/restricted-access/test2');

    generateFilesYamlTestTemplate('Nested toc restricted access', 'mocks/restricted-access/test3');
});
