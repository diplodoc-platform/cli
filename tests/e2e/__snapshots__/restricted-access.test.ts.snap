// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Restricted access > Nested restricted access > filelist 1`] = `
"[
  "plugins/index.md",
  "project/index.md",
  "toc.yaml"
]"
`;

exports[`Restricted access > Nested restricted access 1`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
restricted-access:
  - - admin
    - manager
  - - userA
    - userB
vcsPath: plugins/index.md
sourcePath: plugins/index.md
---
"
`;

exports[`Restricted access > Nested restricted access 2`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
restricted-access:
  - - userA
    - userB
vcsPath: project/index.md
sourcePath: project/index.md
---
"
`;

exports[`Restricted access > Nested restricted access 3`] = `
"title: Yandex Flavored Markdown
items:
  - name: Plugins
    restricted-access:
      - admin
      - manager
    items:
      - name: Overview
        href: plugins/index.md
        restricted-access:
          - userA
          - userB
  - name: Organizing a YFM project
    items:
      - name: Overview
        href: project/index.md
        restricted-access:
          - userA
          - userB
path: toc.yaml
"
`;

exports[`Restricted access > Nested toc restricted access > filelist 1`] = `
"[
  "a1.md",
  "folder1/a1.md",
  "folder1/folder2/a1.md",
  "toc.yaml"
]"
`;

exports[`Restricted access > Nested toc restricted access 1`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
vcsPath: a1.md
sourcePath: a1.md
---
This is the /a1.md file content.
"
`;

exports[`Restricted access > Nested toc restricted access 2`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
restricted-access:
  - - userA
    - userB
  - - userAtoc2
    - userBtoc2
vcsPath: folder1/a1.md
sourcePath: folder1/a1.md
---
This is the /folder1/a1.md file content.
"
`;

exports[`Restricted access > Nested toc restricted access 3`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
restricted-access:
  - - userA
    - userB
  - - userAtoc3
    - userBtoc3
vcsPath: folder1/folder2/a1.md
sourcePath: folder1/folder2/a1.md
---
This is the /folder1/folder1/a1.md file content.
"
`;

exports[`Restricted access > Nested toc restricted access 4`] = `
"items:
  - name: A1
    href: a1.md
  - name: folder1
    restricted-access:
      - userA
      - userB
    items:
      - name: A1
        href: folder1/a1.md
        restricted-access:
          - userAtoc2
          - userBtoc2
      - name: folder1
        items:
          - name: A1
            href: folder1/folder2/a1.md
            restricted-access:
              - userAtoc3
              - userBtoc3
path: toc.yaml
"
`;

exports[`Restricted access > Simple restricted access > filelist 1`] = `
"[
  "index.md",
  "plugins/index.md",
  "plugins/index2.md",
  "plugins/index3.md",
  "plugins/index4.md",
  "toc.yaml"
]"
`;

exports[`Restricted access > Simple restricted access 1`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
restricted-access:
  - - admin
vcsPath: index.md
sourcePath: index.md
---
"
`;

exports[`Restricted access > Simple restricted access 2`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
restricted-access:
  - - admin
vcsPath: plugins/index.md
sourcePath: plugins/index.md
---
"
`;

exports[`Restricted access > Simple restricted access 3`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
restricted-access:
  - - admin
  - - admin
    - user
vcsPath: plugins/index2.md
sourcePath: plugins/index2.md
---
"
`;

exports[`Restricted access > Simple restricted access 4`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
restricted-access:
  - - admin
vcsPath: plugins/index3.md
sourcePath: plugins/index3.md
---
"
`;

exports[`Restricted access > Simple restricted access 5`] = `
"---
metadata:
  - name: generator
    content: Diplodoc Platform vDIPLODOC-VERSION
restricted-access:
  - - admin
  - - customInFile
vcsPath: plugins/index4.md
sourcePath: plugins/index4.md
---
"
`;

exports[`Restricted access > Simple restricted access 6`] = `
"title: Yandex Flavored Markdown
items:
  - name: Yandex Flavored Markdown
    href: index.md
    restricted-access: admin
  - name: Plugins
    items:
      - name: Overview
        href: plugins/index.md
        restricted-access:
          - admin
  - name: Plugins 2
    restricted-access: admin
    items:
      - name: Overview 2
        href: plugins/index2.md
        restricted-access:
          - admin
          - user
      - name: Overview 3
        href: plugins/index3.md
        restricted-access:
          - admin
      - name: Overview 4
        href: plugins/index4.md
        restricted-access:
          - admin
path: toc.yaml
"
`;
